### PR TITLE
Add pls integration design doc

### PR DESCRIPTION
**Pull request by jwtty**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
Initial design document for Azure PLS integration for loadBalancer typed services.
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: site/content/en/development/design-docs/pls-integration.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+| `service.beta.kubernetes.io/azure-pls-auto-approval`                     | `"sub1 sub2 sub3 … subN"`          | A space separated list of Azure subscription ids. This allows PE connection requests from the subscriptions listed to the PLS to be automatically approved. |  Optional | `[]` |
+
+For more details about each configuration, please refer to [Azure Private Link Service Documentation](https://docs.microsoft.com/en-us/cli/azure/network/private-link-service?view=azure-cli-latest#az-network-private-link-service-create).
+
+
+## Design Details
+
+### Creating AKS managed PrivateLinkService
+
+When a `LoadBalancer` typed service is created, a LB frontend IP configuration is created with either a dynamically generated or user specified IP. Or if the service has `loadBalancerIP` in its spec, an existing LB frontend IP configuration may be reused. When service has `service.beta.kubernetes.io/azure-pls-create` set to `true`, a PLS resource attached to the LB frontend is created in the `MC_` resource group for the AKS cluster.
```
Comment (feiskyer) [inline site/content/en/development/design-docs/pls-integration.md:39]: for updated service (e.g. by adding the PLS annotations), this should also be supported?

File: site/content/en/development/design-docs/pls-integration.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+For more details about each configuration, please refer to [Azure Private Link Service Documentation](https://docs.microsoft.com/en-us/cli/azure/network/private-link-service?view=azure-cli-latest#az-network-private-link-service-create).
+
+
+## Design Details
+
+### Creating AKS managed PrivateLinkService
+
+When a `LoadBalancer` typed service is created, a LB frontend IP configuration is created with either a dynamically generated or user specified IP. Or if the service has `loadBalancerIP` in its spec, an existing LB frontend IP configuration may be reused. When service has `service.beta.kubernetes.io/azure-pls-create` set to `true`, a PLS resource attached to the LB frontend is created in the `MC_` resource group for the AKS cluster. 
+
+The Kubernetes service creating the PLS is assigned as the owner of the resource. Azure cloud provider tags the PLS with service name `<namespace>\<service name>`. Only the owner service can later update the properties of the PLS resource.
```
Comment (feiskyer) [inline site/content/en/development/design-docs/pls-integration.md:41]: for the tag mentioned here, could you also add its key here, e.g. `"k8s-azure-service"`?

File: site/content/en/development/design-docs/pls-integration.md (hunk 1 above)
Comment (jwtty) [reply site/content/en/development/design-docs/pls-integration.md]: Good catch! Thanks for pointing out. Updated!

File: site/content/en/development/design-docs/pls-integration.md (hunk 2 above)
Comment (jwtty) [reply site/content/en/development/design-docs/pls-integration.md]: I take a look at how we name the PIP labels: `kubernetes-dns-label-service: default\nginx`, so I'm now following similar pattern, `kubernetes-owner-service: default\nginx`. Updated doc.

**Review by feiskyer — COMMENTED** [review]
Thanks for adding this

/lgtm
/approve

File: site/content/en/development/design-docs/pls-integration.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+## PrivateLinkService annotations
+
+Below is a list of annotations supported for Kubernetes services with Azure PLS created:
+
+| Annotation | Value | Description | Required | Default |
+| ------------------------------------------------------------------------ | ---------------------------------- | ------------------------------------------------------------ |------|------|
+| `service.beta.kubernetes.io/azure-pls-create`                            | `"true"`                           | Boolean indicating whether a PLS needs to be created. | Required | |
+| `service.beta.kubernetes.io/azure-pls-name`                              | `<PLS name>`                       | String specifying the name of the PLS resource to be created. | Optional | `"pls-<LB frontend config id>"` |
+| `service.beta.kubernetes.io/azure-pls-ip-configuration-subnet`           |`<Subnet name>`                     | String indicating the subnet to which the PLS will be deployed.  This subnet must exist in the same VNET as the backend pool. | Required | |
```
[the remark this answers is not in the document]
Comment (jwtty) [reply site/content/en/development/design-docs/pls-integration.md:23]: Thanks, making it clear NAT IPs are chosen from this subnet.

File: site/content/en/development/design-docs/pls-integration.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+For more details about each configuration, please refer to [Azure Private Link Service Documentation](https://docs.microsoft.com/en-us/cli/azure/network/private-link-service?view=azure-cli-latest#az-network-private-link-service-create).
+
+
+## Design Details
+
+### Creating AKS managed PrivateLinkService
+
+When a `LoadBalancer` typed service is created, a LB frontend IP configuration is created with either a dynamically generated or user specified IP. Or if the service has `loadBalancerIP` in its spec, an existing LB frontend IP configuration may be reused. When a service is created with annotation `service.beta.kubernetes.io/azure-pls-create` set to `true` or updated later with the annotation added, a PLS resource attached to the LB frontend is created in the `MC_` resource group for the AKS cluster. 
+
+The Kubernetes service creating the PLS is assigned as the owner of the resource. Azure cloud provider tags the PLS with service name `kubernetes-owner-service: <namespace>\<service name>`. Only the owner service can later update the properties of the PLS resource.
```
[the remark this answers is not in the document]
Comment (jwtty) [reply site/content/en/development/design-docs/pls-integration.md:41]: Good catch! I use surface for work and MAC personally. This is a mistake I make almost everyday.

**Pull request merged**